svc-events/events/evtsubscription.go

```diff
@@ -898,11 +900,30 @@ func validateFields(request *evmodel.RequestBody) (int32, string, []interface{},
 		}
 	}
 	if request.EventFormatType == "" {
-		request.EventFormatType = evmodel.EventFormatType
-	} else if request.EventFormatType == "MetricReport" {
-		return http.StatusBadRequest, errResponse.PropertyMissing, []interface{}{"EventFormatType"}, fmt.Errorf("Unsupported EventFormatType")
-	} else if request.EventFormatType != evmodel.EventFormatType {
-		return http.StatusBadRequest, errResponse.PropertyMissing, []interface{}{"EventFormatType"}, fmt.Errorf("Invalid EventFormatType")
+		request.EventFormatType = "Event"
+	}
+
+	if _, ok := validEventFormatTypes[request.EventFormatType]; !ok {
+		return http.StatusBadRequest, errResponse.PropertyValueNotInList, []interface{}{request.EventFormatType, "EventFormatType"}, fmt.Errorf("Invalid EventFormatType")
+	}
+
+	if len(request.EventTypes) == 0 && request.EventFormatType == "MetricReport" {
+		request.EventTypes = []string{"MetricReport"}
+	}
+
+	for _, eventType := range request.EventTypes {
+		if _, ok := validEventTypes[eventType]; !ok {
+			return http.StatusBadRequest, errResponse.PropertyValueNotInList, []interface{}{request.EventTypes, "EventTypes"}, fmt.Errorf("Invalid EventTypes")
+		}
+	}
+
+	if request.EventFormatType == "MetricReport" {
+		if len(request.EventTypes) > 1 {
+			return http.StatusBadRequest, errResponse.PropertyValueFormatError, []interface{}{"EventTypes", "EventType"}, fmt.Errorf("Unsupported EventType")
```

Comment:
Please update MessageArgs in response with request.EventTypes instead of "EventType"

Reply:
Done